packages/auth-plugins/README.md

```diff
@@ -8,6 +8,120 @@ Authentication plugins for Deephaven. Used by [AuthBootstrap](../app-utils/src/c
 npm install --save @deephaven/auth-plugins
 ```
 
+## Developing New Auth Plugins
+
+Export an `AuthPlugin` from a module to register an authentication plugin. Authentication plugins must implement the [AuthPlugin interface](./src/AuthPlugin.ts#L32). Authentication plugins can display a UI which then triggers how to login.
+
+The Web UI loads all plugins on initialization, and uses the first available authentication plugin for authenticating. A sequence diagram of this flow at a high level, where `AuthPlugin` is the first authentication plugin that is available.
+
+```mermaid
+sequenceDiagram
+  participant U as User
+  participant W as Web UI
+  participant S as Server
+  participant P as AuthPlugin
+  participant J as JS API
+  U->>W: Open app
+  W->>S: Load plugin modules
+  S-->>W: PluginModule[]
+  W->>P: Login
+  P->>J: client.login()
+  J-->>P: Login success
+  P-->>W: Login success
```

Comment:
Maybe add something like this for denoting how an auth plugin is chosen? I'm not sure. Kind of tricky to chart some of this

Maybe call AuthPlugin something like Selected AuthPlugin or Supported AuthPlugin? First Supported AuthPlugin?

```
W->>S: get auth types
S-->>W: AuthTypes[]
```